### PR TITLE
GHA Support

### LINKED ISSUE #24: GHA config file React Frontend
Create template Ci/CD config file

Parent Issue: https://github.com/commitdev/zero/issues/238

### PR DESCRIPTION
**Pull request by dtoki**
Closes #24 
**TODO:**
Currently wrapping up the approval workflow.

### CODE REVIEW

File: templates/.github/workflows/prod-pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      # Run Tests
+        - name: Run Unit Tests
+          uses: actions/setup-node@v1
+          with: 
+            node-version: 12.x
+        - run: |
+            yarn
+            yarn test --watchAll=false
+      # Build Code
+        - name: Build Static Site
```
Comment (bmonkman) [inline templates/.github/workflows/prod-pipeline.yml:29]: The production pipeline shouldn't have a build step. Both staging and production should share an artifact.

File: templates/.github/workflows/stage-pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+          name: stage-build-artifact
+          path: build/
+# TODO: Deploy for stage
+  deploy-stage:
+    name: Upload to Amazon S3
+    runs-on: ubuntu-latest
+    needs: build-stage
+    env:
+      region: us-west-2
+      bucket: <% bucket %>
```
Comment (bmonkman) [inline templates/.github/workflows/stage-pipeline.yml:49]: This is not a valid template var

File: templates/.github/workflows/stage-pipeline.yml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      # Configure Aws account for S3 Sync
+      - name: Configure AWS credentials for S3 sync
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ env.region }}
+      
+      # Deploy to s3
+      - name: Copy files to the test website with the AWS CLI
```
Comment (bmonkman) [inline templates/.github/workflows/stage-pipeline.yml:65]: "test website"

File: templates/.github/workflows/stage-pipeline.yml (hunk 2 above)
Comment (dtoki) [reply templates/.github/workflows/stage-pipeline.yml:49]: right caught that forgot to push my latest changes.

File: templates/.github/workflows/stage-pipeline.yml (hunk 3 above)
Comment (dtoki) [reply templates/.github/workflows/stage-pipeline.yml:65]: right same as ☝🏽 forgot to push changes..

**Pull request closed without merging**